mountpoint-s3/src/data_cache/express_data_cache.rs

```diff
@@ -247,4 +275,37 @@ mod tests {
             "cache entry returned should match original bytes after put"
         );
     }
+
+    #[tokio::test]
+    async fn large_object_bypassed() {
+        let bucket = "test-bucket";
+        let config = MockClientConfig {
+            bucket: bucket.to_string(),
+            part_size: 8 * 1024 * 1024,
+            enable_backpressure: true,
+            initial_read_window_size: 8 * 1024 * 1024,
+            ..Default::default()
+        };
+        let client = Arc::new(MockClient::new(config));
+        let cache = ExpressDataCache::new(
+            bucket,
+            client,
+            "unique source description",
+            1024 * 1024,
+            EXPRESS_CACHE_MAX_OBJECT_SIZE,
+        );
+        let data_1 = vec![0u8; 1024 * 1024 + 1];
+        let data_1 = ChecksummedBytes::new(data_1.into());
+        let cache_key_1 = ObjectId::new("a".into(), ETag::for_tests());
+        // PUT and GET for a large object should be no-op
+        cache
+            .put_block(cache_key_1.clone(), 0, 0, data_1.clone(), data_1.len())
+            .await
+            .expect("cache should be accessible");
+        let get_result = cache
+            .get_block(&cache_key_1, 0, 0, data_1.len())
+            .await
+            .expect("cache should be accessible");
+        assert!(get_result.is_none());
```

Comment:
Should we check the client as well here?

Reply:
added `assert_eq!(client.object_count(), 0, "cache must be empty");`